Jobs/Statistics/LoadMetricsDataJob.php

```diff
@@ -0,0 +1,112 @@
+        $statsUniqueItemInvestigationsDao = DAORegistry::getDAO('UsageStatsUniqueItemInvestigationsTemporaryRecordDAO'); /* @var UsageStatsUniqueItemInvestigationsTemporaryRecordDAO $statsUniqueItemInvestigationsDao */
+        $statsUniqueItemRequestsDao = DAORegistry::getDAO('UsageStatsUniqueItemRequestsTemporaryRecordDAO'); /* @var UsageStatsUniqueItemRequestsTemporaryRecordDAO $statsUniqueItemRequestsDao */
+        $statsInstitutionDao = DAORegistry::getDAO('UsageStatsInstitutionTemporaryRecordDAO'); /* @var UsageStatsInstitutionTemporaryRecordDAO $statsInstitutionDao */
+
+        $statsTotalDao->deleteByLoadId($this->loadId);
+        $statsUniqueItemInvestigationsDao->deleteByLoadId($this->loadId);
+        $statsUniqueItemRequestsDao->deleteByLoadId($this->loadId);
+        $statsInstitutionDao->deleteByLoadId($this->loadId);
+    }
+
+    /**
+     * Load the entries inside the temporary database associated with
+     * the passed load id to the metrics tables.
+     */
+    private function _loadData(): bool
+    {
+        $statsTotalDao = DAORegistry::getDAO('UsageStatsTotalTemporaryRecordDAO'); /* @var UsageStatsTotalTemporaryRecordDAO $statsTotalDao */
+        $statsUniqueItemInvestigationsDao = DAORegistry::getDAO('UsageStatsUniqueItemInvestigationsTemporaryRecordDAO'); /* @var UsageStatsUniqueItemInvestigationsTemporaryRecordDAO $statsUniqueItemInvestigationsDao */
+        $statsUniqueItemRequestsDao = DAORegistry::getDAO('UsageStatsUniqueItemRequestsTemporaryRecordDAO'); /* @var UsageStatsUniqueItemRequestsTemporaryRecordDAO $statsUniqueItemRequestsDao */
+
+        $statsTotalDao->removeDoubleClicks(StatisticsHelper::COUNTER_DOUBLE_CLICK_TIME_FILTER_SECONDS);
+        $statsUniqueItemInvestigationsDao->removeUniqueClicks();
+        $statsUniqueItemRequestsDao->removeUniqueClicks();
+
+        $statsTotalDao->loadMetricsContext($this->loadId);
+        $statsTotalDao->loadMetricsIssue($this->loadId);
+        $statsTotalDao->loadMetricsSubmission($this->loadId);
+
+        // Geo database only contains total and unique investigations (no extra requests differentiation)
+        $statsTotalDao->deleteSubmissionGeoDailyByLoadId($this->loadId); // always call first, before loading the data
+        $statsTotalDao->loadMetricsSubmissionGeoDaily($this->loadId);
+        $statsUniqueItemInvestigationsDao->loadMetricsSubmissionGeoDaily($this->loadId);
+
+        $statsTotalDao->deleteCounterSubmissionDailyByLoadId($this->loadId); // always call first, before loading the data
+        $statsTotalDao->loadMetricsCounterSubmissionDaily($this->loadId);
+        $statsUniqueItemInvestigationsDao->loadMetricsCounterSubmissionDaily($this->loadId);
+        $statsUniqueItemRequestsDao->loadMetricsCounterSubmissionDaily($this->loadId);
+
+        $statsTotalDao->deleteCounterSubmissionInstitutionDailyByLoadId($this->loadId); // always call first, before loading the data
```

Comment:
This is unfortunately not possible, because the methods are upserts, i.e. they either insert or update the data, and 3 of them are used for totals, unique investigations and unique requests

Reply:
OK, but how do I know how to use `$statsTotalDao->loadMetricsCounterSubmissionInstitutionDaily($this->loadId);` correctly? How will I know when I need to call `$statsTotalDao->deleteCounterSubmissionInstitutionDailyByLoadId($this->loadId);` first?